pkg/platform/runtime/buildplan/buildplan.go

```diff
@@ -275,20 +302,49 @@ func buildRuntimeClosureMap(baseID strfmt.UUID, lookup map[strfmt.UUID]interface
 		uniqueDeps = append(uniqueDeps, id)
 	}
 
-	info, err := getSourceInfo(currentArtifact.GeneratedBy, lookup)
-	if err != nil {
-		return errs.Wrap(err, "Could not resolve source information")
-	}
+	if model.IsStateToolArtifact(currentArtifact.MimeType) {
+		info, err := getSourceInfo(currentArtifact.GeneratedBy, lookup)
+		if err != nil {
+			return errs.Wrap(err, "Could not resolve source information")
+		}
 
-	result[strfmt.UUID(currentArtifact.NodeID)] = artifact.Artifact{
-		ArtifactID:       strfmt.UUID(currentArtifact.NodeID),
-		Name:             info.Name,
-		Namespace:        info.Namespace,
-		Version:          &info.Version,
-		RequestedByOrder: true,
-		GeneratedBy:      currentArtifact.GeneratedBy,
-		Dependencies:     uniqueDeps,
-		URL:              currentArtifact.URL,
+		result[strfmt.UUID(currentArtifact.NodeID)] = artifact.Artifact{
+			ArtifactID:       strfmt.UUID(currentArtifact.NodeID),
+			Name:             info.Name,
+			Namespace:        info.Namespace,
+			Version:          &info.Version,
+			RequestedByOrder: true,
+			GeneratedBy:      currentArtifact.GeneratedBy,
+			Dependencies:     uniqueDeps,
+			URL:              currentArtifact.URL,
+			MimeType:         currentArtifact.MimeType,
+		}
+	} else {
+		// For whatever reason we have artifacts whose displayName are prefixed with their node ID, so strip it
```

Comment:
The `displayName` property was added later, hence why it's not used here. I believe they went with a slightly odd naming convention. You might want to verify that the assumptions you're making below are in line with what is outlined here: https://activestatef.atlassian.net/browse/PB-3681

Reply:
Thanks, I've commented on that ticket. It doesn't seem that it was implemented the way that ticket describes.